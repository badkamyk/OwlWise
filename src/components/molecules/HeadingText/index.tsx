import HeadingParagraph from "../../atoms/HeadingParagraph";
import MainHeading from "../../atoms/HeadingMain";
import { useTrail, animated } from "@react-spring/web";
import React from "react";

const Trails = ({ children }: { children: React.ReactNode }) => {
    const items = React.Children.toArray(children);
    const trail = useTrail(items.length, {
        from: { opacity: 0, transform: "translate3d(0,40px,0)" },
        to: { opacity: 1, transform: "translate3d(0,0px,0)" },
        config: { duration: 1200 }
    });
    return (
        <div className="p-3 text-center">
            {trail.map(({ ...style }, index) => (
                <animated.div key={index} style={style}>
                    {items[index]}
                </animated.div>
            ))}
        </div>
    );
};

export default function HeadingText() {
    return (
        <Trails>
            <MainHeading />
            <HeadingParagraph />
        </Trails>

    )
}

