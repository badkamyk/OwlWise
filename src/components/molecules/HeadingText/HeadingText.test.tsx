import { render, screen } from "@testing-library/react";
import { expect, it } from "vitest";
import HeadingText from "./index";

it("renders HeadingText correctly", () => {
	const { container } = render(<HeadingText />);
	expect(container.getElementsByClassName("p-3 text-center")).toBeTruthy();
	expect(screen.getByText("Better Data")).toBeTruthy();
	expect(screen.getByText("Scalable AI.")).toBeTruthy();
	expect(screen.getByText("Check pricing")).toBeTruthy();
	expect(
		screen.getByText(
			"Here at OwlWise we focus on markets where technology, innovation, and capital can unlock long-term value and drive economic growth."
		)
	).toBeTruthy();
	expect(screen.getByRole("link")).toBeTruthy();
	expect(screen.getByRole("link")).toHaveProperty("href", "/pricing");
	expect(container).toMatchSnapshot();
});
