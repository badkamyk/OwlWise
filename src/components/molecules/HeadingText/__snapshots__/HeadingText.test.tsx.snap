// Vitest Snapshot v1

exports[`HeadingText 1`] = `
<div>
  <div
    class="p-3 text-center"
  >
    <div
      style="opacity: 0; transform: translate3d(0,40px,0);"
    >
      <h2
        class="mb-3 text-3xl font-extrabold text-gray-900 dark:text-white md:text-5xl lg:text-6xl"
      >
        <span
          class="bg-gradient-to-r from-sky-400 to-emerald-600 bg-clip-text text-transparent"
        >
          Better Data
        </span>
         
        Scalable AI.
      </h2>
    </div>
    <div
      style="opacity: 0; transform: translate3d(0,40px,0);"
    >
      <p
        class="mb-3 text-lg font-normal text-gray-500 dark:text-gray-400 lg:text-xl"
      >
        Here at OwlWise we focus on markets where technology, innovation, and capital can unlock long-term value and drive economic growth.
      </p>
    </div>
    <div
      style="opacity: 0; transform: translate3d(0,40px,0);"
    >
      <button
        class="group relative mb-2 mr-2 inline-flex items-center justify-center overflow-hidden rounded-lg bg-gradient-to-br from-green-400 to-blue-600 p-0.5 text-sm font-medium text-gray-900 hover:text-white focus:outline-none focus:ring-4 focus:ring-green-200 group-hover:from-green-400 group-hover:to-blue-600 dark:text-white dark:focus:ring-green-800"
      >
        <a
          class="relative rounded-md bg-white px-5 py-2.5 transition-all duration-75 ease-in group-hover:bg-opacity-0 dark:bg-gray-900"
          href="/pricing"
        >
          Check pricing
        </a>
      </button>
    </div>
  </div>
</div>
`;
